app/src/androidTest/java/org/mozilla/fenix/helpers/TestHelper.kt

```diff
@@ -389,20 +392,31 @@ object TestHelper {
 
     /**
      * Changes the default language of the entire device, not just the app.
+     * Only works on the Debug version, not allowed to change system settings in other production builds.
      * Runs the test in its testBlock.
-     * Cleans up and sets the default locale after it's are done.
+     * Cleans up and sets the default locale after it's done.
      */
     fun runWithSystemLocaleChanged(locale: Locale, testRule: ActivityTestRule<HomeActivity>, testBlock: () -> Unit) {
-        val defaultLocale = Locale.getDefault()
+        if (BuildConfig.DEBUG) {
```

Comment:
I looked into better handling build variants. Ideally we should have set-up our structure to place variant tests inside the variant folder (`src/variant`), as within `src/androidTest/`root they are shared across variants and that's probably not what we want these days.

https://developer.android.com/studio/test/advanced-test-setup#create-instrumented-test-for-build-variant

I'm not sure if there's a way to filter annotate the test with this variant check so it its not executed by the runner 

I looked at https://developer.android.com/reference/androidx/test/filters/package-summary and don't see anything. I guess this is suffice?

Reply:
I think you want `Config.isDebug`.

`BuildConfig.DEBUG` is whether the app is "[debuggable](https://developer.android.com/guide/topics/manifest/application-element#debug)" and not the debug _variant_. It's confusing unfortunately.

Reply:
done. Thanks!